src/runtime/components/data/Table.vue

```diff
@@ -152,7 +152,7 @@ export default defineComponent({
       default: undefined
     }
   },
-  emits: ['update:modelValue'],
+  emits: ['update:modelValue', 'onSort'],
```

Comment:
Shouldn't it be `sort`? I think it's better to use it as such `<UTable @sort="onSort" />`.

Reply:
Maybe `update:sort` is better, because then it could be used with v-model (`v-model:sort=""`)

Reply:
This would be a different feature as this doesn't watch for the input but just emits.

Reply:
For some reason when I tried `v-model:sort` it didn't update the value, so I fell back to just emit.

Reply:
@DarkGhostHunter of course, because v-model is just syntax sugar for setting a prop to some value and then updating that value with the payload of a specific event. In this case there already is a prop called `sort` which takes care of passing the value in from the parent. You added an event to be able to pass the value back out of the component to its parent. https://vuejs.org/guide/components/v-model.html
What I'm suggesting is naming that event in a way (`update:sort`), so that the prop `sort` can be used with v-model.

Reply:
@Blechlawine I did that:

- add `update:sort` emit
- run `emit('update:sort', sort.value)` son sorting
- use `v-model:sort="reactive"` on the component
- use `const sort = reactive({ column: null, direction: 'desc' })`.

No changes. Did I miss something?

Reply:
@DarkGhostHunter in that case the sort variable can't be defined with reactive, because afaik, v-model sets the value, and doesn't just set the individual properties. And with `reactive` the entire value can't be reassigned. You would need to use `ref({ column ... })`

Reply:
Update: If I use `v-model:order` and move everything there, while using a `ref` on the reactive element, it works flawlessly.

Reply:
Yep, using `const sort = reactive(...)` doesn't work. You're right.

Reply:
Fixed. You can use `v-model:sort="refElement"` if I use `update:sort`. I'll update the PR.

Reply:
And it should also work with just the event: `@update:sort="doSomething"`

Reply:
Didn't test that but it should.